src/players/shmup/components/patterns.rs

```diff
@@ -0,0 +1,75 @@
+use bevy::prelude::*;
+use rand::prelude::*;
+
+// static mut RNG :ThreadRng = rand::thread_rng();
+
+/// Pattern1 defines a movement pattern
+#[derive(Component, Debug)]
+// We could define better names
+pub struct Pattern1 {
+    pub speed: f32,
+}
+
+/// Pattern2 defines a movement pattern
+#[derive(Component, Debug)]
+pub struct Pattern2 {
+    pub speed: f32,
+}
+
+/// Pattern3 defines a movement pattern
+#[derive(Component, Debug)]
+pub struct Pattern3 {
+    pub speed: f32,
+    /// Indicate when should change direction
+    pub timer: Timer,
+    pub dir_x: f32,
+    pub dir_y: f32,
+}
+
+pub fn pattern3_random_dir(p: &mut Pattern3) {
```

Comment:
Why is there no dir update in the 0..20 range ?
Is it because you want 40% probability to keep the same dir and 20% probability to change to each other dir ?